lib/frame.js

```diff
@@ -443,3 +567,57 @@ function _addFrameOutput(parent, property, output) {
     parent.push(output);
   }
 }
+
+/**
+ * Node matches if it is a node, and matches the pattern as a frame.
+ *
+ * @param state the current framing state.
+ * @param pattern used to match value
+ * @param value to check
+ * @param flags the frame flags.
+ */
+function _nodeMatch(state, pattern, value, flags) {
+  if(!('@id' in value)) {
+    return false;
+  }
+  const nodeObject = state.subjects[value['@id']];
+  return nodeObject && _filterSubject(state, nodeObject, pattern, flags)
+}
+
+/**
+ * Value matches if it is a value and matches the value pattern
+ *
+ * * `pattern` is empty
+ * * @values are the same, or `pattern[@value]` is a wildcard, and
+ * * @types are the same or `value[@type]` is not null
+ *   and `pattern[@type]` is `{}`, or `value[@type]` is null
+ *   and `pattern[@type]` is null or `[]`, and
+ * * @languages are the same or `value[@language]` is not null
+ *   and `pattern[@language]` is `{}`, or `value[@language]` is null
+ *   and `pattern[@language]` is null or `[]`.
+ *
+ * @param pattern used to match value
+ * @param value to check
+ */
+function _valueMatch(pattern, value) {
+  const v1 = value['@value'];
+  const t1 = value['@type'];
+  const l1 = value['@language'];
+  const v2 = pattern['@value'] || [];
+  const t2 = pattern['@type'] || [];
+  const l2 = pattern['@language'] || [];
+
+  if(v2.length === 0 && t2.length === 0 && l2.length === 0) {
+    return true;
+  }
+  if(!(v2.includes(v1) || types.isEmptyObject(v2[0]))) {
```

Comment:
The phantomjs test failed here in some tests where v2 and v1 are the string "P".  Every other test case has v2 as an array.  Based on default above and usage here it looks like that might be a bug?  If not the 'core-js/fn/string/includes' helper would have to be added to main and karma webpack configs.

Reply:
Yeah, that’s a bug.